### PR TITLE
organize screening, put z1 z2 related pow vars to screen_factors_t

### PR DESCRIPTION
**Pull request by zhichen3**
addresses issue #1396

### CODE REVIEW

**Comment by zingale** [comment]
test suite: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2023-11-28-003/index.html

those diffs look to be roundoff

**Comment by yut23** [comment]
It might be good to pull out the magic number values for SCREEN_METHOD into named constants (e.g. `#define SCREEN_METHOD_null 0`, `#define SCREEN_METHOD_chabrier1998 4`, etc.). Not sure if that should go here or another PR

File: screening/screen_data.H [hunk 1]
```diff
@@ -17,20 +17,51 @@ namespace scrn {
...
+      amrex::Real zs53 = 0.0;
+      amrex::Real z1_53 = 0.0;
+      amrex::Real z2_53 = 0.0;
       amrex::Real zs13 = 0.0;
       amrex::Real zs13inv = 0.0;
       amrex::Real zhat = 0.0;
       amrex::Real zhat2 = 0.0;
       amrex::Real lzav = 0.0;
       amrex::Real aznut = 0.0;
       amrex::Real ztilde = 0.0;
```
Comment (yut23) [inline screening/screen_data.H:44]: ztilde looks unused in screen5

Comment (zhichen3) [reply screening/screen_data.H]: fixed

**Review by yut23 — APPROVED** [review]
I'll do the magic number refactor once this gets merged.

**Pull request merged**